src/main/java/seedu/parser/EventParser.java

```diff
@@ -56,48 +55,48 @@ private void splitByEventFlags(String[] tokens) throws PacException {
         for (String token : tokens) {
             if (token.length() < 2) {
                 if (mostRecent == null) {
-                    throw new PacException("EventParser: Flag is too short");
+                    throw new PacException("Flag is too short.");
                 } else if (validFlagToAppend(mostRecent)) {
                     append(mostRecent, token);
                 }
             } else {
                 switch (token.substring(0, 2).toLowerCase()) {
                 case "n/":
-                    ensureNotDuplicateFlag(name, "EventParser: Duplicate name flag");
+                    ensureNotDuplicateFlag(name, "name");
                     name += token.substring(2);
                     mostRecent = "name";
                     break;
                 case "t/":
-                    ensureNotDuplicateFlag(time, "EventParser: Duplicate time flag");
+                    ensureNotDuplicateFlag(time, "time");
                     time += token.substring(2);
                     mostRecent = "time";
                     break;
                 case "d/":
-                    ensureNotDuplicateFlag(date, "EventParser: Duplicate date flag");
+                    ensureNotDuplicateFlag(date, "date");
                     date += token.substring(2);
                     mostRecent = "date";
                     break;
                 case "v/":
-                    ensureNotDuplicateFlag(venue, "EventParser: Duplicate venue flag");
+                    ensureNotDuplicateFlag(venue, "venue");
                     venue += token.substring(2);
                     mostRecent = "venue";
                     break;
                 case "i/":
-                    ensureNotDuplicateFlag(index, "EventParser: Duplicate index flag");
+                    ensureNotDuplicateFlag(index, "index");
                     try {
                         index = Integer.parseInt(token.substring(2));
                     } catch (NumberFormatException m) {
```

Comment:
Perhaps can write this try catch block in a different method for cleaner code?

Reply:
Just a suggestion (:

Reply:
The `try... catch` block is quite short though... If I refactor this, then I feel also should refactor other cases also. But other cases only take 2 lines.